Source/ASDisplayNode.h

```diff
@@ -191,6 +191,11 @@ AS_EXTERN NSInteger const ASDefaultDrawingPriority;
  */
 - (void)setLayerBlock:(ASDisplayNodeLayerBlock)layerBlock;
 
+/**
+ * Get the context for this node, if one was set during creation.
+ */
+@property (nullable, readonly) ASNodeContext *nodeContext;
```

Comment:
Do we need to expose the `nodeContext` publicly?

Reply:
To your question above, if users want to implement custom containers, and they want to add new nodes to a given context outside of our automatic places (like layout) they will need to push the context for an existing node. At least that's my thinking although I haven't written out an example case yet.